application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js

```diff
@@ -1 +1,34 @@
-//this is the namesapce activating this component
\ No newline at end of file
+//this is the namespace for the functions for this component
+
+; (function (ns, document) {
+    "use strict";
+    
+    function init() {
+        Array.from(document.querySelectorAll("#ticker")).forEach(($component) => {
+            // parse json value from data-model attribute as component config
+            const componentConfig = JSON.parse($component.getAttribute("data-model"));
+            const componentTopic = componentConfig.websocketTopic;
+            const componentHost = componentConfig.websocketHost;
+            const componentDataSource = componentConfig.dataSource;
+
+            console.log("[ticker - behavior.js] - Ticker Component");
+            // TMS.
+            if (componentHost && componentTopic) {
+                $component.setAttribute("id", componentTopic);
+                ns.tmsConnected(componentHost, componentTopic, $component);
+            }
+            // JSON
+            else if (componentDataSource) {
+                ns.jsonConnected(componentDataSource, $component);
+            }
+            // MODEL 
+            else {
+                ns.modelDataConnected($component);
+            }
+        });
+    }
+    
+    $(function () {
+        init();
+    });
```

Comment:
what is this!? cant this be executed from behaviour, on some document ready event?

Reply:
This is getting triggered in the behaviour.js only and getting started on ready.

This is the short end of the document.onReady in jquery